From: Sven Roederer <devel-sven@geroedel.de>
Date: Mon, 31 May 2021 23:05:26 +0200
Subject: wizard: use "redirect_landingurl" feature of mod-freifunk to bring up wizard on 1st-boot

The luci-mod-freifunk has the new feature to redirect to a custom page when LuCI is called
without an explicit page (https://github.com/freifunk/openwrt-packages/commit/21e69cd27d736f13295e2af79108617ccc694b45). Let's
use this to redirect to the wizard and remove the redirection when the wizard is finished.
This way we don't need the "redirect on firstboot" patch (up to Kathleen 0.1.2) and can
also drop our forked "luci-mod-freifunk-ui". "luci-mof-freifunk-ui" was basically a fork
of "luci-mof-freifunk" with the mentioned patch included.

Signed-off-by: Sven Roederer <devel-sven@geroedel.de>

diff --git a/utils/luci-app-ffwizard-berlin/luasrc/view/freifunk/assistent/reboot.htm b/utils/luci-app-ffwizard-berlin/luasrc/view/freifunk/assistent/reboot.htm
index 71cf6fadfc64cc515bf34b4ec061edb1356dded4..97dcfdec14d9b3a7eb985363f1f7c368443a9925 100644
--- a/utils/luci-app-ffwizard-berlin/luasrc/view/freifunk/assistent/reboot.htm
+++ b/utils/luci-app-ffwizard-berlin/luasrc/view/freifunk/assistent/reboot.htm
@@ -1,4 +1,10 @@
 <%
+  -- remove landingpage redirection of luci-mod-freifunk before final reboot
+  local uci = require "luci.model.uci".cursor()
+  uci:set('freifunk', 'luci', 'redirect_landingurl', '')
+  uci:commit('freifunk', 'luci')
+  uci:save('freifunk')
+
   luci.sys.reboot()
 %>
 
diff --git a/utils/luci-app-ffwizard-berlin/root/etc/uci-defaults/wizarddefaults b/utils/luci-app-ffwizard-berlin/root/etc/uci-defaults/wizarddefaults
index 9c850a03d993d5b42c82a5b271728d42a020f251..b0714f097c58a7af2875f4d6a31c019fe927cdff 100644
--- a/utils/luci-app-ffwizard-berlin/root/etc/uci-defaults/wizarddefaults
+++ b/utils/luci-app-ffwizard-berlin/root/etc/uci-defaults/wizarddefaults
@@ -6,3 +6,9 @@ uci set freifunk-policyrouting.pr.enable=1
 uci set freifunk-policyrouting.pr.fallback=0
 uci set freifunk-policyrouting.pr.zones="freifunk"
 uci commit freifunk-policyrouting
+
+# redirect to wizard when it has never run before
+[ $(uci -q get ffwizard.settings.runbefore) = 'true' ] && exit 0
+uci set freifunk.luci=setting
+uci set freifunk.luci.redirect_landingurl='/cgi-bin/luci/admin/freifunk/assistent?luci_username=root&luci_password='
+uci commit freifunk.luci
